server/graphql/v1/mutations/orders.js

```diff
@@ -231,6 +231,43 @@ const hasPaymentMethod = order => {
   }
 };
 
+export const deletePaymentMethod = async (collective, fingerprint, transaction) => {
+  // Check if the credit card is already saved
+  const oldPaymentMethod = await models.PaymentMethod.findOne({
+    where: {
+      CollectiveId: collective.id,
+      service: 'stripe',
+      saved: true,
```

Comment:
Just a note that, if someone has a saved credit card and fills out the number for the same card while unchecking "Save for future payments" it will remove the already-saved card from payment methods.

That's arguably a good approach.